### PR TITLE
Add a spark operator

### PR DESCRIPTION
**Pull request by holdenk**
This adds the spark operator as a manifest. This is my first attempt at writing kustomize, so feedback would be really appreciated.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/174)

### CODE REVIEW

**Comment by holdenk** [comment]
Hi @jlewi / @richardsliu would you be ok giving this ok-to-test while I'm finishing it up? If not no worries.

**Comment by holdenk** [comment]
cc @richardsliu if you have some cycles I'd really appreciate your feedback. I did get this working but I'm a little fuzzy on the best practices and how to go about improving the tests (if required).

File: spark-operator/base/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# Value of this field is prepended to the
+# names of all resources
+namePrefix: spark-operator-spark
+
+# List of resource files that kustomize reads, modifies
+# and emits as a YAML string
+resources:
+- spark-operator-spark-sa.yaml
+- spark-operator-sparkoperator-cr-clusterrole.yaml
```
Comment (richardsliu) [inline spark-operator/base/kustomization.yaml]: Do we need to repeat "sparkoperator" in the file name here? The directory name already contains spark-operator, so I think these can all be omitted.

You can checkout the Pytorch one as an example: https://github.com/kubeflow/manifests/tree/master/pytorch-job

Also note that most of the manifests follow a 2-level structure - top level for "package" and lower level for "components".

File: spark-operator/base/spark-operator-sparkoperator-cr-clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: spark-operator-sparkoperator-cr
```
Comment (richardsliu) [inline spark-operator/base/spark-operator-sparkoperator-cr-clusterrole.yaml]: Same comment for these names.

File: spark-operator/base/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: kubeflow
+# Labels to add to all resources and selectors.
+commonLabels:
+  kustomize.component: spark-operator
+  app.kubernetes.io/instance: spark-operator
+  app.kubernetes.io/managed-by: Tiller
```
Comment (swiftdiaries) [inline spark-operator/base/kustomization.yaml]: Small nit:
Can we remove references to helm charts here?

File: spark-operator/base/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  app.kubernetes.io/name: sparkoperator
+  helm.sh/chart: sparkoperator-0.2.4
```
Comment (swiftdiaries) [inline spark-operator/base/kustomization.yaml]: Same here

File: spark-operator/base/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+# Images modify the tags for images without
+# creating patches.
+images:
+- name: gcr.io/spark-operator/spark-operator
+  newTag: v2.4.0-v1beta1-0.8.2
+
+# Value of this field is prepended to the
+# names of all resources
+namePrefix: spark-operator-spark
```
Comment (swiftdiaries) [inline spark-operator/base/kustomization.yaml]: Just `spark-operator` would do

**Comment by jlewi** [comment]
@holdenk If you send a PR my way adding yourself to the org
https://github.com/kubeflow/internal-acls/blob/master/github-orgs/kubeflow/org.yaml

I'll be more then happy to LGTM it so you can get an invite to the org. Thanks for all your contributions.

I'll let @richardsliu and @swiftdiaries review this PR.

/assign @richardsliu 
/assign @swiftdiaries

**Comment by richardsliu** [comment]
Can you also add an application overlay? Check out some examples here: https://github.com/kubeflow/manifests/tree/master/tf-training/tf-job-operator/overlays/application

**Comment by holdenk** [comment]
Sure

**Pull request merged**